src/ThreeEditor/js/viewport/Viewport.js

```diff
@@ -26,544 +26,542 @@ const callWithHidden = (object, fn) => {
 	object.visible = visible;
 };
 
-export function Viewport(
-	name,
-	editor,
-	{ objects, grid, planeHelpers, selectionBox },
-	{
-		orthographic,
-		cameraPosition,
-		cameraUp,
-		clipPlane,
-		planePosLabel,
-		planeHelperColor,
-		showPlaneHelpers = false,
-		gridRotation
-	} = {}
-) {
-	this.name = name;
-
-	const {
-		figureManager: scene,
-		zoneManager,
-		detectorManager,
-		sceneHelpers,
-		specialComponentsManager,
-		signals,
-		contextManager
-	} = editor;
-
-	const config = {
-		selectFigures: true,
-		selectZones: false,
-		selectGeometries: false,
-		visible: false
-	};
-
-	const sceneViewHelpers = new THREE.Scene();
-
-	const container = new UIPanel();
-	container.setId('ViewPanel');
-	container.setPosition('absolute');
-	container.setOverflow('hidden');
-	container.setWidth('100%');
-	container.setHeight('100%');
-	container.dom.setAttribute('tabindex', '0');
-
-	const wrapperDiv = new UIDiv();
-	wrapperDiv.setPosition('relative');
-	wrapperDiv.add(container);
-
-	const canvas = document.createElement('canvas');
-	container.dom.appendChild(canvas);
-
-	const context = canvas.getContext('2d');
-
-	const cameraPersp = new THREE.PerspectiveCamera(50, 1, 0.001, 10000);
-	cameraPersp.name = 'Perspective';
-	cameraPersp.position.copy(cameraPosition ?? new THREE.Vector3(0, 20, 20)); // default camera position other than (0,0,0) to see anything
-
-	const cameraOrtho = new THREE.OrthographicCamera(-4, 4, 4, -4, 0.001, 10000);
-	cameraOrtho.name = 'Orthographic';
-	cameraOrtho.position.copy(cameraPersp.position);
-	cameraOrtho.zoom = 0.2;
-
-	// in clipping plane views only Orthographic camera is used, hence is "up" axis adjustment is required we do so
-	cameraUp && cameraOrtho.up.copy(cameraUp);
-
-	const cameras = [cameraOrtho, cameraPersp];
-
-	let camera = orthographic ? cameraOrtho : cameraPersp;
-	updateAspectRatio();
-
-	Object.defineProperty(this, 'camera', {
-		get() {
-			return camera;
-		},
-		set(value) {
-			const position = camera.position.clone();
-
-			camera = value;
-
-			updateCamera(camera, position);
-		}
-	});
-
-	if (!orthographic) container.add(new ViewportCamera(this, cameras));
-
-	const viewHelper = new ViewHelper(camera, container);
-	viewHelper.disabled = orthographic;
-
-	let viewClipPlane = null;
-
-	if (clipPlane) {
-		viewClipPlane = new ViewportClippedViewCSG(
-			name,
-			editor,
-			this,
-			planeHelpers,
-			zoneManager.zoneContainer.children,
-			signals.zoneGeometryChanged,
-			signals.zoneAdded,
-			signals.zoneRemoved,
-			wrapperDiv.dom,
-			{
-				clipPlane,
-				planeHelperColor,
-				planePosLabel
-			}
-		);
-	}
+export class Viewport {
```

Comment:
that code is mostly copy-pasted from https://threejs.org/ (or https://github.com/mrdoob/three.js/)

see https://github.com/yaptide/ui/blob/1ef735fc943d97d729349d5c07db5e895b47c909/src/ThreeEditor/js/viewport/Viewport.js#L16

maybe instead of adjusting the code yourself you go to https://github.com/mrdoob/three.js/ and ask mrdoob what he thinks of using his library in vite ?  

you will achieve more with communication than with coding !